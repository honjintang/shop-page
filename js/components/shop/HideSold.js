import React from 'react'
import { connect } from 'react-redux'
import ReusableButton from '~/components/general/reusableButton'
import { toggleSoldItems } from '~/actions'
import { isShowingSoldItems } from '~/reducers/display'

export const HideSold = props => {
    const { toggleSoldItems, showSoldItems } = props

    return  (
        <ReusableButton 
            id='hide-show'
            label='Hide Sold Items'
            onClick={() => toggleSoldItems(!showSoldItems)}
            active={!showSoldItems}
        />
    )
}

const mapStateToProps = state => ({
    showSoldItems: isShowingSoldItems(state),
})

const mapDispatchToProps = {
    toggleSoldItems,
}

export default connect(mapStateToProps, mapDispatchToProps)(HideSold)