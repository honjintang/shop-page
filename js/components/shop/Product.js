import React from 'react'
import Card from 'react-bootstrap/Card'

const Product = ({id, img, title, brand, price, size, sold }) => (
        <Card id={id}>
            <Card.Img variant="top" src={img} />
            <Card.Body>
            <Card.Title>{title}</Card.Title>
            <Card.Text>
                <p>Brand: {brand || 'N/A'}</p>
                <p>Size: {size}</p>
                <p>Price: Â£{price}</p>
            </Card.Text>
            </Card.Body>
        </Card>
)

export default Product