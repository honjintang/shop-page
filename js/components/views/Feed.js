import React from 'react'
import { connect } from 'react-redux'
import Card from 'react-bootstrap/Card'
import { getProducts } from '~/actions'
import { getAllProducts } from '~/reducers/products'

export const Feed = props => {
    const { products, getProducts } = props
    React.useEffect(() => {
        getProducts()
    }, [getProducts])

    return  (
        <>
            {products &&
                <Card.Header>
                    <h1>products</h1>
                </Card.Header>
            }
        </>
    )
}

const mapStateToProps = state => ({
    products: getAllProducts(state),
})

const mapDispatchToProps = {
    getProducts,
}

export default connect(mapStateToProps, mapDispatchToProps)(Feed)
