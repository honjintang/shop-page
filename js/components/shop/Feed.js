import React from 'react'
import { connect } from 'react-redux'
import CardColumns from 'react-bootstrap/CardColumns'
import Product from '~/components/shop/Product'
import { getProducts } from '~/actions'
import { getAllProducts } from '~/reducers/products'

export const Feed = props => {
    const { products, getProducts } = props
    React.useEffect(() => {
        getProducts()
    }, [getProducts])

    return  (
        <>
            {products &&
                <CardColumns>
                    {products.map(p => <Product key={p.id} {...p} />)}
                </CardColumns>
            }
        </>
    )
}

const mapStateToProps = state => ({
    products: getAllProducts(state),
})

const mapDispatchToProps = {
    getProducts,
}

export default connect(mapStateToProps, mapDispatchToProps)(Feed)
