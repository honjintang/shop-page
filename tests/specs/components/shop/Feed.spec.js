import React from 'react'
import { shallow } from 'enzyme'
import { Feed } from '~/components/shop/Feed'

describe('components/shop/Feed', () => {
    const mockGetProducts= jest.fn()
    const mockProducts = [{ id: 'mock_id_1', title: 'mock_title_1' }, { id: 'mock_id_2', title: 'mock_title_2' }]
    let useEffect
    let component

    const mockUseEffect = () => {
        useEffect.mockImplementationOnce(f => f());
    }

    beforeEach(() => {
        useEffect = jest.spyOn(React, 'useEffect')
        mockUseEffect()
        component = shallow(<Feed getProducts={mockGetProducts} products={mockProducts}/>)
    })

    it('useEffect fetches products', () => {
        expect(mockGetProducts).toHaveBeenCalledTimes(1)
    })

    it('renders all products', () => {
        expect(component).toMatchSnapshot()
    })
})