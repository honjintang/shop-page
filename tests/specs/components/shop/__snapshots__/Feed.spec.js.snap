// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/shop/Feed renders all products 1`] = `
<Fragment>
  <CardColumns>
    <Product
      id="mock_id_1"
      key="mock_id_1"
      title="mock_title_1"
    />
    <Product
      id="mock_id_2"
      key="mock_id_2"
      title="mock_title_2"
    />
  </CardColumns>
</Fragment>
`;
